lib/services/downloads_service.dart

```diff
@@ -681,15 +683,56 @@ class DownloadsService {
     // Step 4 - Make sure there are no unanchored nodes in metadata.
     _downloadsLogger.info("Starting downloads repair step 4");
     downloadCounts[repairStepTrackingName] = 4;
+    final Map<int, LyricDto?> idsWithLyrics = HashMap();
+    var allItems = _isar.downloadItems.where().findAllSync();
+    for (var item in allItems) {
+      if (item.baseItem?.mediaStreams?.any((stream) => stream.type == "Lyric") ?? false) {
+        // check if lyrics are already downloaded
+        if (_isar.downloadedLyrics.where().isarIdEqualTo(item.isarId).countSync() > 0) {
+          continue;
+        }
+        idsWithLyrics[item.isarId] = null;
+      }
+    }
+    //TODO move this into a separate function in the background service
+    final JellyfinApiHelper _jellyfinApiData = GetIt.instance<JellyfinApiHelper>();
+    for (var id in idsWithLyrics.keys) {
+      var canonItem = _isar.downloadItems.getSync(id);
```

Comment:
You should probably just call this item.  We're not in a transaction.

Reply:
I initially misunderstood what you meant, but that's probably a good thing, since I'm now using one loop less ^^